### ISSUE TITLE
Little inconsistency when using effective-name and custom pod-label-selector

### ISSUE THREAD

**Issue by mbuccini** [akka/akka-management#167]
Hello everyone,

I was digging into the code to understand how `cluster-bootstrap` is connected to `discovery-kubernetes-api`, and I found out that we have the following:

https://github.com/akka/akka-management/blob/master/discovery-kubernetes-api/src/main/scala/akka/discovery/kubernetes/Settings.scala#L26

```
  def podLabelSelector(name: String): String =
    kubernetesApi.getString("pod-label-selector").format(name)
```

which is based on the assumption that the config file (https://github.com/akka/akka-management/blob/master/discovery-kubernetes-api/src/main/resources/reference.conf) may have a placeholder:
```
akka.discovery {
  kubernetes-api {
    pod-label-selector = "app=%s"
```
If this is the case, it will resolve to `akka.management.cluster.bootstrap.contact-point-discovery.effective-name` (which is built based on `<service-name>.<service-namespace>`, unless it's set differently). This happens because if we use `akka.discovery.method=kubernetes-api`, when we call `discovery.lookup(serviceName, ...)` (here: https://github.com/akka/akka-management/blob/master/cluster-bootstrap/src/main/scala/akka/management/cluster/bootstrap/dns/HeadlessServiceDnsBootstrap.scala#L205), it will do the following:

```
  def lookup(name: String, resolveTimeout: FiniteDuration): Future[Resolved] =
    for {
      token <- apiToken()

      labelSelector = settings.podLabelSelector(name)

      _ = system.log.info("Querying for pods with label selector: [{}]", labelSelector)

      request <- optionToFuture(podRequest(token, settings.podNamespace, labelSelector),
        s"Unable to form request; check Kubernetes environment (expecting env vars ${settings.apiServiceHostEnvName}, ${settings.apiServicePortEnvName})")
```
meaning that we may have an inconsistency. 

**Long story short**: if we set  

```akka.discovery {
  kubernetes-api {
     pod-label-selector = "app=my-custom-label"
```

and at the same time we set:

```
akka.management.cluster.bootstrap.contact-point-discovery.effective-name=my-cool-service-name
```

The following line (https://github.com/akka/akka-management/blob/master/cluster-bootstrap/src/main/scala/akka/management/cluster/bootstrap/dns/HeadlessServiceDnsBootstrap.scala#L126):

```override def receive: Receive = {
    case InitiateBootstrapping ⇒
      log.info("Locating service members; Lookup [{}]. Using discovery [{}], join decider [{}]", serviceName,
        discovery.getClass.getName, joinDecider.getClass.getName)
      lookup()

      context become bootstrapping(sender())
  }
```

will log
```
Locating service members; Lookup "my-cool-service-name"
```
while it's actually trying to lookup the pods with label "app=my-custom-label".

**Comment by ktoso** [comment]
Thanks for reporting; So, you propose to fix the log statement? 
Would you send in a PR please?

**Comment by mbuccini** [comment]
Will do. It's a tiny change (it's already in my fork, but I need to sign the CLA ...)

**Comment by ktoso** [comment]
Solved in #172
